fix(tests): create a team, to ensure projects can be seen

klochek opened the pull request:
Without creating and joining a team, a user cannot see a list of projects.  You can see this if you turn off headless when running the test: creating projects and navigating back to the project dashboard will show no projects.  Within the affected test, this means we always see the "Remain Calm" dashboard, even if we have existing projects.  Creating and joining a team corrects this behaviour, and (I believe) fixes the test flakiness.

priscilawebdev commented:
@klochek Thank you for working on this test and trying to improve it 🙏

Just to clarify the intended behavior: when a user creates a project and then navigates back - either by clicking "Back to Platform Selection" or using the browser’s back button - we silently delete the project on purpose. This is based on the assumption that the user wants to choose a different platform.

Because of this, we always expect a "remain calm" screen to be displayed in that scenario, and the current test correctly reflects that behavior.

Separately, when creating a new team in the app, the creator is usually added as a member by default. However, the test currently doesn’t assign the user to the team, which might be contributing to the flakiness.

To fix this, you can add `members=[self.user]` when creating the team. For example:

`self.create_team(organization=self.org, name="team three", members=[self.user])`

This might help stabilize the test. Let’s try that and see if the flakiness goes away.

klochek committed:
fix(tests): create a team, to ensure projects can be seen

Without creating and joining a team, a user cannot see a list of projects. You can see this if you turn off headless when running the test: creating projects and navigating back to the project dashboard will show no projects. Within the affected test, this means we always see the "Remain Calm" dashboard, even if we have existing projects. Creating and joining a team corrects this behaviour, and (I believe) fixes the test flakiness.

priscilawebdev reviewed (approved):
🙌

klochek committed:
Merge branch 'master' into christopherklochek/rtc-1030-flaky-test-test_project_deletion_on_going_back
